lib/utils/bootstrap-ui-package.ts

```diff
@@ -0,0 +1,65 @@
+import {sync as commandExistsSync} from 'command-exists';
+import {join} from 'path';
+import {SpawnSyncStrict} from './spawn-sync-strict';
+
+const defaultSpawnOptions = {stdio: 'inherit', encoding: 'utf-8'} as const;
+
+export function bootstrapUIPackage(answers: any, context) {
+  const isGitInstalled = commandExistsSync('git');
+
+  /** Helper to exit process in case one sub-process fails */
+  const spawnSyncStrict = SpawnSyncStrict(({status}, a, b) => {
+    context.log.error(
+      `Bootstrapping process failed because sub-task "${a} ${b.join(
+        ' ',
+      )}" did not succeed (status ${status})`,
+    );
+    return 1;
+  });
+
+  /* Initialize before npm i because husky requires git for setup */
+  if (isGitInstalled) {
+    context.log.info(`Git is installed, initializing repo.`);
+    spawnSyncStrict('git', ['init'], defaultSpawnOptions);
+  }
+
+  /* Some files should be automatically generated. Set them to read-only to avoid user manipulation */
+  if (commandExistsSync('chmod')) {
```

Comment:
This section raises some red flags for me. Why do we need to mess with file permissions?

Reply:
There will be a new functionality in `shell-ui` soon: `lsc modules add` command. It will generate `routes.ts` and `menu-items.ts` files with data derived from a custom manifest (`labshare-manifest.js`). 
It works like similarly to how a `package-lock.json` is generated from `package.json`.
Since `routes.ts` and `menu-items.ts` look just like any other typescript file, chances are developers will try to edit it manually. So I left some comments in the files and marked them as read-only for extra safety.
**Files:**
* [menu-items.ts](https://github.com/LabShare/lsc/pull/177/files#diff-818d057f8b9fc5f0abad38482806bae2)
* [module-routes.ts](https://github.com/LabShare/lsc/pull/177/files#diff-18f26bd6ccde7f102dee7dbc5d2ee3e3)

The read-only permission is not strictly necessary, though.

Reply:
I see, I would suggest not to modify file permissions and simply provide a warning in the comments. This would be consistent with other scaffolding/generators such as Symfony Flex in PHP.